Code/Include/ROS2/ROS2Bus.h

```diff
@@ -15,15 +15,38 @@
 
 namespace ROS2
 {
+    //! Interface to the central ROS2SystemComponent.
+    //! Use this API through ROS2Interface, for example:
+    //! @code
+    //! auto node = ROS2Interface::Get()->GetNode();
+    //! @endcode
     class ROS2Requests
     {
     public:
         AZ_RTTI(ROS2Requests, "{a9bdbff6-e644-430d-8096-cdb53c88e8fc}");
         virtual ~ROS2Requests() = default;
 
-        // Put your public methods here
+        //! Get a central ROS2 node of the Gem.
+        //! You can use this node to create publishers and subscribers.
+        //! @return The central ROS2 node which holds default publishers for core topics such as /clock and /tf.
+        //! @note Alternatively, you can use your own node along with an executor.
         virtual std::shared_ptr<rclcpp::Node> GetNode() const = 0;
+
+        //! Acquire current time as ROS2 timestamp.
+        //! Timestamps provide temporal context for messages such as sensor data.
+        //! @code
+        //! auto message = sensor_msgs::msg::PointCloud2();
+        //! message.header.stamp = ROS2Interface::Get()->GetROSTimestamp();
+        //! @endcode
+        //! @return Simulation time in ROS2 format. Time source is also valid with non-real time settings.
+        //! @note Make sure to set the use_sim_time parameter for ROS2 nodes which will use the simulation data.
         virtual builtin_interfaces::msg::Time GetROSTimestamp() const = 0;
+
+        //! Send transformation between ROS2 frames.
+        //! @param isDynamic controls whether a static or dynamic transform is sent. Static transforms are published
+        //! only once and are to be used when the spatial relationship between two frames does not change.
+        //! @note Transforms are already published by each ROS2FrameComponent.
+        //! Use this function directly only when default behavior of ROS2FrameComponent is not sufficient.
         virtual void BroadcastTransform(const geometry_msgs::msg::TransformStamped& t, bool isDynamic) const = 0;
```

Comment:
Looks like the @param for the `t` parameter is missing.

Reply:
that was on purpose. It has no additional context except what is already documented here: https://docs.ros2.org/latest/api/geometry_msgs/msg/TransformStamped.html. But this is also true for many other ros2 structs and types. I thought referring users to ros2 documentation as a part of top level docs would be better than providing refs for each of them in the comments, but I also think this could be up for a discussion.

Reply:
Well from outside perspective, I think users would have hard time determining what the TransformStamped parameter should be set to.